### ISSUE TITLE
the use of mathjs 11.3.2 generates a build error : TransferFunction not defined in types/index.d.ts line 291

### ISSUE THREAD

**Issue by dsteve** [josdejong/mathjs#2835]
**Describe the bug**
Building an application package with mathjs 11.3.2 generates the following error : 
```
Error: node_modules/mathjs/types/index.d.ts:291:21 - error TS2304: Cannot find name 'TransferFunction'.
291     ): FunctionNode<TransferFunction, TArgs>
```      
... unless version 4.8.4 of package typescript is used (as TransferFunction is defined in lib/lib.dom.d.ts)

**To Reproduce**
Build a package with mathjs 11.3.2 and use typescript package in version 4.3.5 for instance.

**Note**
I doubt that the resolution of TransferFunction by the typescript package is the expected result as the TransferFunction type in lib/lib.dom.d.ts is : `type TransferFunction = "hlg" | "pq" | "srgb";`

I rather think that Tfn should have been used instead : 
```
  interface FunctionNodeCtor {
    new <TFn = SymbolNode, TArgs extends MathNode[] = MathNode[]>(
      fn: TFn,
      args: TArgs
    ): FunctionNode<TFn, TArgs>
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    onUndefinedFunction: (name: string) => any
  }
```
Also the type of args (set to SymbolNode in node_modules/mathjs/types/index.d.ts) seems unexpected. I would rather think of TArgs or MathNode[] instead (as suggested above).

But I am not an expert in mathjs. So I may be absolutely wrong.
However, I cloned the project, did the changes above, npm run build and npm run test without errors.

**Comment by josdejong** [comment]
Thanks for reporting this issue! I see you traced it back to the following PR: https://github.com/josdejong/mathjs/pull/2733#pullrequestreview-1167449108, that is helpful.

Anyone able to help work out a fix for the type definitions?

**Comment by josdejong** [comment]
This issue has been fixed now in v11.3.3 via 0a810abba3920411e68ffa76d4750b37fd903872, thanks @dsteve for figuring this out.
